web3/account.py

```diff
@@ -0,0 +1,194 @@
+
+from collections import (
+    Mapping,
+)
+import functools
+import json
+import os
+import sys
+
+from cytoolz import (
+    compose,
+)
+
+from eth_keyfile import (
+    create_keyfile_json,
+    decode_keyfile_json,
+)
+
+from eth_keys import (
+    KeyAPI,
+    keys,
+)
+from eth_keys.exceptions import (
+    ValidationError,
+)
+
+from eth_utils import (
+    keccak,
+    is_dict,
+    is_string,
+)
+
+from web3.module import Module
+
+from web3.utils.datastructures import (
+    AttributeDict,
+)
+from web3.utils.encoding import (
+    hexstr_if_str,
+    text_if_str,
+    to_bytes,
+    to_decimal,
+    to_hex,
+    to_hex_with_size,
+)
+from web3.utils.exception import (
+    raise_from,
+)
+from web3.utils.signing import (
+    LocalAccount,
+    annotate_transaction_with_chain_id,
+    sign_message_hash,
+    sign_transaction_dict,
+    signature_wrapper,
+    to_standard_v,
+)
+from web3.utils.transactions import (
+    Transaction,
+    vrs_from,
+)
+
+
+class Account(Module):
+    _keys = keys
+
+    def create(self, extra_entropy=''):
+        extra_key_bytes = text_if_str(to_bytes, extra_entropy)
+        key_bytes = keccak(os.urandom(32) + extra_key_bytes)
+        if sys.version_info.major < 3:
+            key_bytes = to_hex(key_bytes)
+        return self.privateKeyToAccount(key_bytes)
+
+    def decrypt(self, keyfile_json, password):
+        if is_string(keyfile_json):
+            keyfile = json.loads(keyfile_json)
```

Comment:
Feels like this shouldn't throw a decode error if I pass in either a random string or bad JSON.  It'd be nice to issue a better error if decoding fails.

Reply:
Other note.  `is_string` will return `True` for a `bytes` value but I believe `json.loads` will fail in python 3 if you pass it a byte string.

Reply:
If I passed an invalid json string into the parameter `keyfile_json`, I would expect a JSONDecodeError describing where in the string it failed to decode the json. What do you think would be more helpful to say here?

Ah, thanks. Passing in `bytes` would be an odd thing to try, but I'll check for it.

Reply:
> What do you think would be more helpful to say here?

Maybe just the text of the `JSONDecodeError` with a *header* Something like:

```
Error parsing keyfile JSON:

<JSONDecodeError-text-here>
```